google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/CreateCampaignExperiment.java

```diff
@@ -0,0 +1,168 @@
+    CreateCampaignExperimentParams params = new CreateCampaignExperimentParams();
+    if (!params.parseArguments(args)) {
+
+      // Either pass the required parameters for this example on the command line, or insert them
+      // into the code here. See the parameter class definition above for descriptions.
+      params.customerId = Long.parseLong("INSERT_CUSTOMER_ID_HERE");
+      params.baseCampaignId = Long.valueOf("INSERT_CAMPAIGN_ID_HERE");
+    }
+
+    GoogleAdsClient googleAdsClient;
+    try {
+      googleAdsClient = GoogleAdsClient.newBuilder().fromPropertiesFile().build();
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+          "Failed to load GoogleAdsClient configuration from file. Exception: %s%n", fnfe);
+      return;
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      return;
+    }
+
+    try {
+      new CreateCampaignExperiment()
+          .runExample(googleAdsClient, params.customerId, params.baseCampaignId, params.draftId);
+    } catch (GoogleAdsException gae) {
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      System.err.printf(
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /** Runs the example. */
+  private void runExample(
+      GoogleAdsClient googleAdsClient, long customerId, long baseCampaignId, long draftId) {
+    // Creates the campaign experiment.
+    CreateCampaignExperimentMetadata metadata =
+        createExperiment(googleAdsClient, customerId, baseCampaignId, draftId);
+
+    // Logs the created resource name.
+    System.out.printf(
+        "Created campaign experiment with resource name: %s%n", metadata.getCampaignExperiment());
+
+    // Prints additional details about the status of the experiment.
+    printExperimentDetails(googleAdsClient, customerId, metadata);
+  }
+
+  /**
+   * Creates an experiment by issuing an asynchronous operation and immediately blocking to get the
+   * result.
+   */
+  private CreateCampaignExperimentMetadata createExperiment(
+      GoogleAdsClient googleAdsClient, long customerId, long baseCampaignId, long draftId) {
+    // Defines the experiment to be created.
+    CampaignExperiment experiment =
+        CampaignExperiment.newBuilder()
+            .setCampaignDraft(
+                StringValue.of(ResourceNames.campaignDraft(customerId, baseCampaignId, draftId)))
+            .setName(StringValue.of("Campaign experiment = " + UUID.randomUUID()))
```

Comment:
Most of our other examples follow the pattern: "Name #" + System.currentTimeMillis()